packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -200,6 +204,17 @@ class FlutterManifest {
     return null;
   }
 
+  /// Like [supportedPlatforms], but only returns the valid platforms that are supported in flutter plugins.
+  Map<String, dynamic> get validSupportedPlatforms {
```

Comment:
What if we moved this logic into `_validateMultiPlatformYaml` so it fails validation if you add an unsupported platform, instead of silently dropping it on the ground?

Reply:
When the plugin is created with no `--platforms` flag, we explicitily added an invalid platform to make the pubspec format valid. 
https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/plugin/pubspec.yaml.tmpl#L40
If we throw in `_validateMultiPlatformYaml`, it would break.